src/user/formats/hal.ts

```diff
@@ -1,14 +1,18 @@
 import { PrivilegeMap } from '../../privilege/types.ts';
-import { Principal, Group, User, PrincipalIdentity } from '../../types.ts';
+import { Principal, Group, User, PrincipalIdentity, PaginatedResult } from '../../types.ts';
 import { HalResource } from 'hal-types';
 import { LazyPrivilegeBox } from '../../privilege/service.ts';
 import { UserNewResult } from '../../api-types.ts';
 
-export function collection(users: User[], embeddedUsers: HalResource[]): HalResource {
+export function collection(embeddedUsers: HalResource[], paginatedResult: PaginatedResult<User>): HalResource {
+
+  const { principals: users, page: currentPage, total, pageSize, hasNextPage } = paginatedResult;
+
+  const totalPages = Math.ceil(total / pageSize);
 
   const hal: HalResource = {
     _links: {
-      'self': { href: '/user' },
+      'self': { href: `/user?page=${currentPage}` },
```

Comment:
Could generating the string `/user=${currentPage}` be done in a new function such as:

```javascript
function getUserPageHref(result: PaginatedResult<User>): HalLink {
  if (result===1) return { href: '/user' };
  return { href: `/user?page=${currentPage}` };
}
```

This ensures that we're always consistent with generating the link, and the `self` link lines up with the URL which is useful for caching.